### PR TITLE
update to 2018 edition

### PR DESCRIPTION
**Pull request by evanrichter**
Mostly automated with `cargo fix --edition` and `cargo fix --edition-idioms`, and `cargo fmt` as well

### CODE REVIEW

**Comment by evanrichter** [comment]
also fixed all the clippy warnings

File: src/torrent/picker/mod.rs [hunk 1]
```diff
@@ -204,18 +204,24 @@ impl Picker {
...
-            .collect();
-        dl.sort_by_key(|&(_, ref req)| req.num_reqd);
-        for (block, req) in dl {
-            req.rereq(peer.id(), peer.rank);
-            return Some(*block);
-        }
-        None
+            .fold(None, |c: Option<(&Block, &mut Request)>, this| match &c {
+                Some(min) => {
+                    if this.1.num_reqd < min.1.num_reqd {
```
Comment (evanrichter) [inline src/torrent/picker/mod.rs:213]: @Luminarys please review this change before merging, I wasn't confident I could test this thoroughly to make sure it is equivalent.
clippy warned about the usage of a for loop that doesn't actually loop, so I refactored to this, and also took out the intermediate vector.
I'm pretty sure this picks the lowest `num_reqd` as intended, but I'm only 80% confident

File: src/torrent/picker/mod.rs [hunk 2]
```diff
@@ -204,18 +204,24 @@ impl Picker {
...
-            .downloading
+        self.downloading
             .iter_mut()
             .filter(|&(_, ref req)| req.num_reqd < MAX_DUP_REQS && !req.has_peer(peer.id()))
             .take(MAX_DL_REREQ)
-            .collect();
-        dl.sort_by_key(|&(_, ref req)| req.num_reqd);
-        for (block, req) in dl {
-            req.rereq(peer.id(), peer.rank);
-            return Some(*block);
```
Comment (evanrichter) [inline src/torrent/picker/mod.rs:216]: for loop returns early here

File: src/torrent/picker/mod.rs (hunk 1 above)
Comment (Luminarys) [reply src/torrent/picker/mod.rs:213]: This looks correct to me, thanks!

**Comment by Luminarys** [comment]
Can you figure out what's breaking travis? I think at the very least we need to update it to 1.38 min (some crate in the last update requires it).

**Comment by Luminarys** [comment]
Thanks for making this change!

**Comment by evanrichter** [comment]
I forgot to commit the package lock file, fixed now.
I was able to reproduce the build error from CI, and it was a dependency of reqwest (needed by sycli) that needed to update to 1.38.0. Would you like to split the MSRV of synapse-bt and sycli?

**Comment by Luminarys** [comment]
I think we can keep them in sync. I generally like to have the minimum version trail stable by a few months or so so that people who use distribution packages etc. don't have to worry about compilation failures.

**Pull request merged**